new-pipeline/src/auth.js

```diff
@@ -0,0 +1,118 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+(function () {
+
+// The state we used when asking for the token. A 16-character random string.
+// We could use the actual app state if we wanted to.
+let gRequestState;
+
+/* Copied with implicit license from Auth0 docs: randomString */
+function randomString(length) {
```

Comment:
There's really no available library function?

Reply:
Even if there were, we already have a problem of too many JS files on load.